sc-memory/sc-core/sc_memory_context_manager.c

```diff
@@ -156,29 +377,68 @@ void _sc_memory_context_manager_register_user_events(sc_memory_context_manager *
 {
   sc_helper_resolve_system_identifier(
       s_memory_default_ctx, "concept_authentication_request_user", &manager->concept_authentication_request_user_addr);
+  _sc_context_set_access_levels_for_element(
+      manager->concept_authentication_request_user_addr, SC_CONTEXT_ACCESS_LEVEL_TO_ALL_ACCESS_LEVELS);
+
   sc_helper_resolve_system_identifier(
       s_memory_default_ctx, "concept_authenticated_user", &manager->concept_authenticated_user_addr);
+  _sc_context_set_access_levels_for_element(
+      manager->concept_authenticated_user_addr, SC_CONTEXT_ACCESS_LEVEL_TO_ALL_ACCESS_LEVELS);
 
-  manager->on_authentication_request_user_subscription = sc_event_new_ex(
+  manager->on_authentication_request_user_subscription = sc_event_with_user_new(
       s_memory_default_ctx,
       manager->concept_authentication_request_user_addr,
       SC_EVENT_ADD_OUTPUT_ARC,
       manager,
       _sc_memory_context_manager_on_authentication_request_user,
       null_ptr);
-  manager->on_unauthentication_request_user_subscription = sc_event_new_ex(
+  manager->on_remove_authenticated_user_subscription = sc_event_with_user_new(
       s_memory_default_ctx,
-      manager->concept_authentication_request_user_addr,
+      manager->concept_authenticated_user_addr,
       SC_EVENT_REMOVE_OUTPUT_ARC,
       manager,
       _sc_memory_context_manager_on_unauthentication_request_user,
       null_ptr);
+
+  sc_helper_resolve_system_identifier(
+      s_memory_default_ctx, "nrel_user_action_class", &manager->nrel_user_action_class_addr);
+  _sc_context_set_access_levels_for_element(
+      manager->nrel_user_action_class_addr, SC_CONTEXT_ACCESS_LEVEL_TO_ALL_ACCESS_LEVELS);
+
+  sc_context_manager_add_basic_action_class_access_levels("read_action_in_sc_memory", SC_CONTEXT_ACCESS_LEVEL_READ);
```

Comment:
I think it's time for us to store the names of all the keynodes on which sc-memory depends in one file

Reply:
Why?